satellite/referrals/users.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package referrals
+
+import (
+	"net/mail"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/satellite/console"
+)
+
+// ErrValidation validation related error class
+var ErrValidation = errs.Class("validation error")
+
+// CreateUser contains information that's necessary for creating a new user through referral program
+type CreateUser struct {
+	FullName      string `json:"fullName"`
+	ShortName     string `json:"shortName"`
+	Email         string `json:"email"`
+	Password      string `json:"password"`
+	ReferralToken string `json:"referralToken"`
+}
+
+// IsValid checks CreateUser validity and returns error describing whats wrong.
+func (user *CreateUser) IsValid() error {
+	var errors []error
```

Comment:
why the error list instead of returning errors individually?

Reply:
I think this is a UX decision. Instead of showing the user one error at a time, we would show them what's wrong with their input from the form all at once